be/src/runtime/fragment_mgr.cpp

```diff
@@ -501,41 +500,32 @@ void FragmentMgr::cancel_worker() {
     LOG(INFO) << "FragmentMgr cancel worker is going to exit.";
 }
 
-
-Status FragmentMgr::fetch_fragment_exec_infos(PFetchFragmentExecInfosResult* result,
-                                              const PFetchFragmentExecInfoRequest* request) {
-    int fragment_id_list_size = request->finst_id_size();
-    for (int i = 0; i < fragment_id_list_size; i++) {
-        const PUniqueId& p_fragment_id = request->finst_id(i);
-        TUniqueId id;
-        id.__set_hi(p_fragment_id.hi());
-        id.__set_lo(p_fragment_id.lo()); 
-        PFragmentExecInfo* info = result->add_fragment_exec_info();
-        PUniqueId* finst_id = info->mutable_finst_id();
-        finst_id->set_hi(p_fragment_id.hi());
-        finst_id->set_lo(p_fragment_id.lo()); 
-
-        bool is_running = false; 
-        std::lock_guard<std::mutex> lock(_lock);
-        {
-            auto iter = _fragment_map.find(id);
-            if (iter == _fragment_map.end()) {
-                info->set_exec_status(PFragmentExecStatus::FINISHED);
-                continue;
+Status FragmentMgr::trigger_profile_report(const PTriggerProfileReportRequest* request) {
+    if (request->instance_ids_size() > 0) {
+        for (int i = 0; i < request->instance_ids_size(); i++) {
+            const PUniqueId& p_fragment_id = request->instance_ids(i);
+            TUniqueId id;
+            id.__set_hi(p_fragment_id.hi());
+            id.__set_lo(p_fragment_id.lo());
+            {
+                std::lock_guard<std::mutex> lock(_lock);
+                auto iter = _fragment_map.find(id);
```

Comment:
auto iter -> auto& iter

Reply:
ok, i will fix it.

Reply:
the for_each of map return Pair class value as function result, so it can't be refered as left-value reference.